openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+            return TextureResult(
+                path=source,
+                file_hash=None,     # todo: unknown texture hash?
+                colorspace=colorspace,
+                transfer_mode=COPY
+            )
+
+        args = []
+        if color_management["enabled"]:
+            config_path = color_management["config"]
+            if not os.path.exists(config_path):
+                raise RuntimeError("OCIO config not found at: "
+                                   "{}".format(config_path))
+
+            render_colorspace = color_management["rendering_space"]
+
+            self.log.info("tx: converting colorspace {0} "
+                          "-> {1}".format(colorspace,
+                                          render_colorspace))
+            args.extend(["--colorconvert", colorspace, render_colorspace])
+            args.extend(["--colorconfig", config_path])
+
+        else:
+            # We can't rely on the colorspace attribute when not in color
+            # managed mode because the collected color space is the color space
+            # attribute of the file node which can be any string whatsoever
+            # but only appears disabled in Attribute Editor. We assume we're
+            # always converting to linear/Raw if the source file is assumed to
+            # be sRGB.
+            # TODO Without color management do we even know we can do
+            #      "colorconvert" and what config does that end up using since
+            #       colorconvert is a OCIO command line flag for maketx.
+            #       Also, Raw != linear?
+            render_colorspace = "linear"
+            if self._has_arnold():
+                img_info = image_info(source)
+                color_space = guess_colorspace(img_info)
+                if color_space.lower() == "sRGB":
+                    self.log.info("tx: converting sRGB -> linear")
+                    args.extend(["--colorconvert", "sRGB", "Raw"])
+                else:
+                    self.log.info("tx: texture's colorspace "
+                                  "is already linear")
+            else:
+                self.log.warning("tx: cannot guess the colorspace, "
+                                 "color conversion won't be "
+                                 "available!")
+
+        # Note: The texture hash is only reliable if we include any potential
+        # conversion arguments provide to e.g. `maketx`
+        hash_args = ["maketx"]
+        hash_args.extend(args)
+        hash_args.extend(self.extra_args)
+        texture_hash = source_hash(source, *hash_args)
+
+        # Exclude these additional arguments from the hashing because
+        # it is the hash itself
+        args.extend([
+            "--sattrib",
+            "sourceHash",
+            texture_hash
+        ])
+
+        # Ensure folder exists
+        destination = os.path.join(staging_dir, "resources", fname + ".tx")
```

Comment:
```suggestion
        destination = os.path.join(staging_dir, "resources",  "{}.tx".format(fname))
```

Reply:
although I'd probably just split this in two to avoid doing the `os.path.dirname` again later:
```
resources_folder = os.path.join(staging_dir, "resources")
if not os.path.exists(resources_folder):
    os.makedirs(resources_folder)

texture_filename = os.path.join(resources_folder, "{}.tx".format(fname))
```

Reply:
And I'd probably move the declaration of `texture_filename`/`destination` closer to where the variable is actually used:
```
texture_filename = os.path.join(resources_folder, "{}.tx".format(fname))
subprocess_args.extend(["-o", texture_filename])
```